[quelinxiao] ip

quelinxiao opened the pull request:
# Bob
Why `you` should use Bob:
> So you can work faster
* Text-based
* Easy to learn
* ~~FAST~~ SUPER FAST to use

Steps to take:
1. Download it [here](https://github.com/quelinxiao/ip)
2. Open the file
3. Start adding your tasks in :smile:

It is completely **FREE!**

Features:
- [x] Manage tasks
- [x] Manage deadlines
- [x] Find

This is the main method:
```
public static void main(String[] args) {
        new Bob("./data/tasks.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

quelinxiao committed:
Tweak greet and exit message

quelinxiao committed:
Add echo function

quelinxiao committed:
Add list function to chatbot

quelinxiao committed:
Add mark and unmark

quelinxiao committed:
Add different types of tasks

quelinxiao committed:
Modify test scripts

quelinxiao committed:
Add error handling

quelinxiao committed:
Add remove function

NewtonKoh commented on src/main/java/Bob.java:
```diff
@@ -31,7 +31,7 @@ else if (input.equals("list")) {
             }
 
-            else if (input.trim().matches("mark|unmark|deadline|todo|event")) {
+            else if (input.trim().matches("mark|unmark|deadline|todo|event|delete")) {
```
Small nit. I believe the else if should be preceded by the closing curly brackets from the if/if else statement above it.

NewtonKoh reviewed (commented):
LGTM. Small nits

tanjiajiajun commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,221 @@
+        String greet = " Hello! I'm Bob.\n"
+                + " What can I do for you?\n";
+
+        String exit = " Bye. Hope to see you again soon!";
```
Consider changing this to a final String as the value of the greeting and exit phrase will not change

tanjiajiajun commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,221 @@
+                        + " Now you have " + taskList.size() + " tasks in the list.");
+            }
+
+            else if (input.startsWith("todo ")) {
```
Having spaces at the end of the string does not look very clean. Maybe can consider removing that.

tanjiajiajun commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,221 @@
+     */
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + by + ")";
```
Clean way to handle output. I did not know that!

tanjiajiajun reviewed (commented):
Looks great to merge. Just small nits

quelinxiao committed:
Add save and clear function

quelinxiao committed:
Add date time formatting to deadline task

jovantanyk commented on data/bob.txt:
Local data files like this should not be pushed onto remote

jovantanyk commented on src/main/java/Bob.java:
Maybe consider separating your code into different classes and packages :)

jovantanyk commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,430 @@
+/*
+ * This class represents a task we want to record.
+ */
+class Task {
```
Different Classes should not be in the same .java file. Consider separating them into dedicated java files

jovantanyk commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,430 @@
+     *
+     * @return A boolean depending on whether the task is done.
+     */
+    public boolean getIsDone() {
```
By java convention, methods that start with get() usually return a Value like a string, integer or array. For booleans you can just use isDone()

jovantanyk commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,430 @@
+
+    /*
+     * A constructor that depicts a new task.
+     */
```
Since Task is never initiated and instead inherited by other subclasses, consider using an abstract class instead.

jovantanyk commented on src/main/java/Bob.java:
```diff
@@ -0,0 +1,430 @@
+        }
+
+        catch (IOException e) {
+            System.out.println(e.getMessage());
```
Very basic way of handling exceptions. Consider creating a new Exception class so you can consolidate your exceptions and have different behavior for different types of Exceptions. This method of doing things just means that when your program really does have an exception, it'll be hard to pinpoint where or how it came from because the try block is so big

jovantanyk reviewed (commented):
Good attempt however please try and separate out your classes, they should not all be in the same java file. There's also almost no use of OOP here, you can create utility classes for handling Ui, Exceptions, Files, instead of storing them all inside methods.

quelinxiao committed:
Split Bob,java into OOP parts

quelinxiao committed:
Group java files to bob package

quelinxiao committed:
Merge branch 'add-gradle-support'

quelinxiao committed:
Add tests for Parser and Task.java

quelinxiao committed:
Add JavaDoc comments

quelinxiao committed:
Tweak code to comply with coding standard

quelinxiao committed:
Add find feature

quelinxiao committed:
Merge branch 'branch-A-CodingStandard'

quelinxiao committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/bob/Bob.java

quelinxiao committed:
Resolve merge conflicts

quelinxiao committed:
Tweak filePath in bob to be specified in main method

quelinxiao committed:
Implement GUI

quelinxiao committed:
Change gradle build to use Launcher as main class

Main class was set to Bob so the GUI would not work

Changing main class to Launcher allows it to be the entry point of the application

quelinxiao committed:
Include assertions in code

quelinxiao committed:
Improve code quality

quelinxiao committed:
Merge pull request #2 from quelinxiao/branch-A-Assertions

Include assertions in code

quelinxiao committed:
Merge pull request #3 from quelinxiao/branch-A-CodeQuality

Improve code quality

quelinxiao committed:
Add product screenshot

quelinxiao committed:
Update user guide
